### PR TITLE
workflows: Add buildbuddy.yaml schema

### PR DESCRIPTION
**Pull request by bduffany**
Add the basic schema for `buildbuddy.yaml`, which will be used to trigger workflows.

I put the test YAML configs in separate files for ease of authoring (my editor keeps trying to insert tabs because it's Go, and I was experiencing errors because of having tabs in the YAML).

---

**Version bump**: Minor

### CODE REVIEW

**Comment by siggisim** [comment]
I'll let @tylerwilliams  review, but just wanted to give this a drive-by "this looks awesome"!

**Comment by bduffany** [comment]
Noice! Added you as a reviewer as a second set of eyes to be sure that this looks good from an ease-of-use perspective.

File: enterprise/server/workflowconf/workflowconf.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
+package workflowconf
```
Comment (tylerwilliams) [inline enterprise/server/workflowconf/workflowconf.go:1]: maybe we ought to make workflow a "container" directory and move workflowconf -> workflow/conf, workflowcmd -> workflow/command, and workflow -> workflow/runner?

File: enterprise/server/workflowconf/test_data/basic.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+actions:
+  - name: "Build and test"
+    triggers:
+      push:
+        branches: [main]
+      pull_request:
+        branches: [main]
+    bazel_commands:
+      - build //...
+      - test //...
```
Comment (tylerwilliams) [inline enterprise/server/workflowconf/test_data/basic.yaml:10]: This looks awesome, really happy where we netted out!

**Review by tylerwilliams — CHANGES_REQUESTED** [review]

File: enterprise/server/workflowconf/test_data/basic.yaml (hunk 2 above)
Comment (bduffany) [reply enterprise/server/workflowconf/test_data/basic.yaml:10]: Same -- nice and simple!

File: enterprise/server/workflowconf/workflowconf_test.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
+package workflowconf
```
[the remark this answers is not in the document]
Comment (bduffany) [reply enterprise/server/workflowconf/workflowconf_test.go]: Done

File: enterprise/server/workflowconf/workflowconf.go (hunk 1 above)
Comment (tylerwilliams) [reply enterprise/server/workflowconf/workflowconf.go:1]: SGTM

**Review by tylerwilliams — APPROVED** [review]
LGTM

**Pull request merged**